src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -112,31 +113,56 @@ public override void Refine(CodeNamespace generatedCode)
             generatedCode,
             x => $"{x.Name}able"
         );
+        RemoveHandlerFromRequestBuilder(generatedCode);
     }
-    private void InsertOverrideMethodForRequestExecutorsAndBuildersAndConstructors(CodeElement currentElement) {
+    
+    protected static void RenameCancellationParameter(CodeElement currentElement){
+        if (currentElement is CodeMethod currentMethod && currentMethod.IsOfKind(CodeMethodKind.RequestExecutor) && currentMethod.Parameters.OfKind(CodeParameterKind.Cancellation) is CodeParameter parameter)
+        {
+            parameter.Name = "ctx";
+            parameter.Description = "Pass a context parameter to the request";
+            parameter.Kind = CodeParameterKind.Cancellation;
+            parameter.Optional = false;
+            parameter.Type.Name = conventions.ContextVarTypeName;
+            parameter.Type.IsNullable = false;
+        }
+        CrawlTree(currentElement, RenameCancellationParameter);
+    }
+    
+    private void RemoveHandlerFromRequestBuilder(CodeElement currentElement)
+    {
+        if (currentElement is CodeClass currentClass)
+        {
+            // remove all response handlers from request builder param
+            if (currentClass.IsOfKind(CodeClassKind.RequestBuilder))
+            {
+                var codeMethods = currentClass.Methods.Where(x => x.Kind == CodeMethodKind.RequestExecutor);
+                foreach (var codeMethod in codeMethods)
+                {
+                    codeMethod.RemoveParametersByKind(CodeParameterKind.ResponseHandler);
+                }
+            }
+        }
```

Comment:
```suggestion
```suggestion
        if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.RequestBuilder))
        {
            // remove all response handlers from request builder param
            var codeMethods = currentClass.Methods.Where(static x => x.Kind == CodeMethodKind.RequestExecutor);
            foreach (var codeMethod in codeMethods)
            {
                codeMethod.RemoveParametersByKind(CodeParameterKind.ResponseHandler);
            }
        }
```
```